browser/brave_wallet/android/brave_wallet_native_worker.cc

```diff
@@ -77,11 +88,103 @@ bool BraveWalletNativeWorker::UnlockWallet(
       base::android::ConvertJavaStringToUTF8(env, password));
 }
 
+base::android::ScopedJavaLocalRef<jstring>
+BraveWalletNativeWorker::RestoreWallet(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jstring>& mnemonic,
+    const base::android::JavaParamRef<jstring>& password) {
+  auto* keyring_controller = GetBraveWalletService()->keyring_controller();
+  auto* keyring = keyring_controller->RestoreDefaultKeyring(
+      base::android::ConvertJavaStringToUTF8(env, mnemonic),
+      base::android::ConvertJavaStringToUTF8(env, password));
+  if (keyring) {
+    keyring->AddAccounts();
+  }
+  return base::android::ConvertUTF8ToJavaString(
+      env, keyring_controller->GetMnemonicForDefaultKeyring());
+}
+
 void BraveWalletNativeWorker::ResetWallet(JNIEnv* env) {
   auto* keyring_controller = GetBraveWalletService()->keyring_controller();
   keyring_controller->Reset();
 }
 
+void BraveWalletNativeWorker::GetAssetPrice(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jstring>& asset) {
+  brave_wallet::BraveWalletService* brave_wallet_service =
+      GetBraveWalletService();
+  if (brave_wallet_service) {
+    auto* asset_ratio_controller =
+        brave_wallet_service->asset_ratio_controller();
+    asset_ratio_controller->GetPrice(
+        base::android::ConvertJavaStringToUTF8(env, asset),
+        base::BindOnce(&BraveWalletNativeWorker::OnGetPrice,
+                       weak_ptr_factory_.GetWeakPtr()));
+  }
+}
+
+void BraveWalletNativeWorker::OnGetPrice(bool success,
+                                         const std::string& price) {
+  JNIEnv* env = base::android::AttachCurrentThread();
+  Java_BraveWalletNativeWorker_OnGetPrice(
+      env, weak_java_brave_wallet_native_worker_.get(env),
+      base::android::ConvertUTF8ToJavaString(env, price), success);
+}
+
+void BraveWalletNativeWorker::GetAssetPriceHistory(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jstring>& asset,
+    const jint timeFrameType) {
+  brave_wallet::BraveWalletService* brave_wallet_service =
+      GetBraveWalletService();
+  if (brave_wallet_service) {
+    auto* asset_ratio_controller =
+        brave_wallet_service->asset_ratio_controller();
+
+    brave_wallet::mojom::AssetPriceTimeframe time_frame;
```

Comment:
This is seriously fragile https://github.com/brave/brave-core/pull/9399/files#diff-dda545139648bcfd1d3a02d90ece90e4dbfe4a343b456954ea7065de3d2f0d15R44 and there's no reason for it because mojo types can be passed directly to java, but even if this wasn't mojo we should at least be using an enum with `java_cpp_enum`. Even if this was the only option (which it's not), there's not even a comment that these two lists need to be kept in sync.

Reply:
@bridriver I have looked at rewards implementation as reference here and I wanted to update it with MOJO in java but I didn't update since native worker would be obsolete soon. I will update it in a branch as ask for your review.